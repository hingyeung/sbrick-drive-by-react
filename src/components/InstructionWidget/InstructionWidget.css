.instruction {
    background-color: cornflowerblue;
    padding: 8px;
    margin: 8px;
}