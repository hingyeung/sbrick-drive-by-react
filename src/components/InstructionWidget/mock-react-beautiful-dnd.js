import React from 'react';

const provided = {draggableProps: {style: undefined}, innerRef: undefined};
const snapshot = {isDraggingOver: undefined};

export const Draggable = ({children}) => {
    return (
        <div>
            Mock Draggable Component
            {children(provided)}
        </div>
    )
};

export const DragDropContext = ({children}) => {
    return (
        <div>
            Mock DragDropContext Component
            {children}
        </div>
    )
};

export const Droppable = ({children}) => {
    return (
        <div>
            Mock DragDropContext Component
            {children(provided, snapshot)}
        </div>
    )
};