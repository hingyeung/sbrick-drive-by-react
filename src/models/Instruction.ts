export interface Instruction {
    displayName: string;
    id: string;
    index: number;
}
