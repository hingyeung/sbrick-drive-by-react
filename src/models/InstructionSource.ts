export interface InstructionSource {
    id: string;
    displayName: string;
    command: string;
}
