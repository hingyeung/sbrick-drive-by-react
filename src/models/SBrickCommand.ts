export enum SBrickCommand {
  forward,
  backward,
  left,
  right
}