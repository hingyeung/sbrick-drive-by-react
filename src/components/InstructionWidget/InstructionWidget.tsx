import * as React from 'react';
import { Draggable } from 'react-beautiful-dnd';
import { Instruction } from '../../models/Instruction';

interface InstructionWidgetProps {
    instruction: Instruction;
}

export default (props: InstructionWidgetProps) => (
    <Draggable draggableId={props.instruction.id as any as string} index={props.instruction.id}>
        {(provided, snapshot) => (
            <div
                ref={provided.innerRef}
                className="instruction"
                style={{...provided.draggableProps.style as object}}
                {...provided.dragHandleProps}
            >
                Instruction
            </div>
        )}
    </Draggable>
);