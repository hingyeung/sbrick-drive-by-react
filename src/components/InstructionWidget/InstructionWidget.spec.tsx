import * as renderer from 'react-test-renderer';
import * as React from 'react';
import { Instruction } from '../../models/Instruction';
import InstructionWidget from './InstructionWidget';
// https://github.com/facebook/jest/issues/936#issuecomment-214556122
// rbd is used to mock 'react-beautiful-dnd'
import * as rbd from 'react-beautiful-dnd';
import { DraggableProvided } from 'react-beautiful-dnd';
import { SBrickCommand } from '../../models/SBrickCommand';

jest.mock('react-beautiful-dnd');

const INSTRUCTION: Instruction = {
  id: 'instructionId',
  displayName: 'displayName',
  sBrickCommand: SBrickCommand.forward
};

describe('InstructionWidget', function () {
  beforeEach(() => {
    const draggableProvided: DraggableProvided = {
      innerRef: () => {
        return;
      },
      draggableProps: {
        style: null,
        'data-react-beautiful-dnd-draggable': ''
      },
      dragHandleProps: null
    };
    // even though tslint is complaining about "attempted to assign to readonly property",
    // it works, and I don't know how to do it any other way.
    // @ts-ignore
    rbd.Draggable = ({children}: any) => (<div>{children(draggableProvided)}</div>);
  });

  it('should render properly', () => {
    const tree = renderer
      .create(
        <InstructionWidget index={0} instruction={INSTRUCTION}/>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it.only('should display instruction displayName', () => {
  //     const tree = renderer.create(
  //             <InstructionWidget index={0} instruction={INSTRUCTION}/>
  //         );
  //     expect(tree).toContain('displayName');
  // });
});