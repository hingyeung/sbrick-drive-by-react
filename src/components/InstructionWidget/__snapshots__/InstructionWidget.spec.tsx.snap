// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionWidget should render properly 1`] = `
<div>
  <div>
    <div
      className="instruction"
      style={Object {}}
    >
      displayName
    </div>
  </div>
</div>
`;
